src/main/java/core/basesyntax/service/TransactionStrategy.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service;
+
+import core.basesyntax.FruitTransaction;
+
+public class TransactionStrategy {
+    public OperationHandler getShopService(FruitTransaction fruit) {
+        switch (fruit.getOperation()) {
+            case BALANCE -> {
+                return new BalanceService();
+            }
+            case SUPPLY -> {
+                return new SupplyService();
+            }
+            case RETURN -> {
+                return new ReturnService();
+            }
+            case PURCHASE -> {
+                return new PurchaseService();
+            }
+            default -> throw new RuntimeException("Unknown operation");
```

Comment:
still, your Strategy pattern implementation is invalid, thats how it should look like:
```suggestion
    private final Map<FruitTransaction.Operation, OperationHandler> operationToHandlerMap;

    public OperationHandler getShopService(FruitTransaction.Operation operation) {
        return operationToHandlerMap.get(operation);
```